### DISCUSSION TITLE
Not using generateStaticParams in LocaleLayout

### DISCUSSION THREAD

**Discussion by huhassan** [amannn/next-intl#383] [Q&A]
Hi, when I include generateStaticParams in LocaleLayout as the following.

```js
export async function generateStaticParams() {
  return ["ar", "en"].map((locale) => ({ locale }));
}
```
It prevents me from using server components (next-auth getServerSession) as children of LocaleLayout and gives the following error.

> Error: Dynamic server usage: headers

I noticed that generateStaticParams as in the above example is used in [example-next-13](https://github.com/amannn/next-intl/tree/main/examples/example-next-13) but it's not used in [example-next-13-next-auth](https://github.com/amannn/next-intl/tree/main/examples/example-next-13-next-auth).

I'm new to nextjs and would appreciate any help in knowing the difference between including and not including generateStaticParams in the mentioned examples. Can I simply remove generateStaticParams?

**Comment by amannn** [comment]
You can learn more about this in the Next.js docs: [Static and Dynamic Rendering](https://nextjs.org/docs/app/building-your-application/rendering/static-and-dynamic-rendering).

next-auth only works with dynamic rendering, therefore you'd have to remove `generateStaticParams` (see [docs](https://nextjs.org/docs/app/api-reference/functions/generate-static-params)).

Hope this helps!
